util/include/util/odbtransaction.h

```diff
@@ -175,12 +175,22 @@ void persistAll(Cont& cont_, std::shared_ptr<odb::database> db_)
         << item->toString();
       LOG(warning)
         << ex.what() << std::endl
-        << "AST nodes in this translation unit will be ignored!";
+        << "Further changes in this transaction will might be ignored!";
     }
     catch (const odb::database_exception& ex)
     {
-      // TODO: Error code should be checked and rethrow if it is not unique
-      // constraint error. Error code may be database specific.
+      LOG(debug) << item->toString();
+
+#ifdef DATABASE_PGSQL
+      if (std::string(ex.what()).find("25P02") != std::string::npos)
```

Comment:
It is fine, because it should occur rarely, but `std::strstr()` could also be used for finding substrings in a C style string.

Reply:
Fixed in d0099ddf824925de4247256b084ca474631f6fc9.